Assets/GothicVR/Scripts/Manager/AudioManager.cs

```diff
@@ -0,0 +1,148 @@
+        }
+
+        private void UpdateAudioSourcesAroundPlayer()
+        {
+            if (!IsAudioListenerObtained)
+            {
+                return;
+            }
+
+            // Calculate the current grid cell position of the audio listener
+            Vector3Int listenerCell = GetGridCellFromPosition(audioListener.transform.position);
+
+            if (listenerCell == playerCell)
+            {
+                return;
+            }
+
+            playerCell = listenerCell;
+
+            // Iterate over the cells around the player in a circular area
+            UpdateAudioSourcesWithinRadius(listenerCell: listenerCell, radius: 3);
+        }
+
+        public void AddAudioSource(GameObject gameObj, AudioSource audioSource)
+        {
+            if (audioSources.ContainsKey(gameObj))
+            {
+                return;
+            }
+            audioSources.Add(gameObj, audioSource);
+
+            // Add the audio source to the appropriate grid cell
+            Vector3Int gridCell = GetGridCellFromPosition(gameObj.transform.position);
+            if (!gridCells.TryGetValue(gridCell, out List<GameObject> audioSourcesInCell))
+            {
+                audioSourcesInCell = new List<GameObject>();
+                gridCells.Add(gridCell, audioSourcesInCell);
+            }
+            audioSourcesInCell.Add(gameObj);
+        }
+
+        public void SetAudible(GameObject gameObj, bool isAudible)
+        {
+            gameObj.SetActive(isAudible);
+        }
+
+        public void SetAudioListener(AudioListener audioListener)
+        {
+            this.audioListener = audioListener;
+        }
+
+        /// <summary>
+        /// Converts a position in world space to the corresponding grid cell position.
+        /// This function divides the position coordinates by the cell size and rounds down to the nearest integer to determine the grid cell coordinates.
+        /// The resulting grid cell position is returned as a Vector3Int.
+        /// </summary>
+        /// <param name="position">The position in world space.</param>
+        /// <returns>The corresponding grid cell position.</returns>
+        private Vector3Int GetGridCellFromPosition(Vector3 position)
+        {
+            int x = Mathf.FloorToInt(position.x / cellSize);
+            int y = Mathf.FloorToInt(position.y / cellSize);
+            int z = Mathf.FloorToInt(position.z / cellSize);
+            return new Vector3Int(x, y, z);
+        }
+
+        /// <summary>
+        /// Updates the audio sources within a specified radius around the listener cell.
+        /// It iterates over each cell within the radius and checks if it falls within the circular area.
+        /// If so, it retrieves the audio sources associated with that cell and determines if they should be audible.
+        /// https://demonstrations.wolfram.com/ApproximatingSpheresWithBoxes/ - graphical visualisation of how the sphere would look like
```

Comment:
Would it make sense to have size of the boxes configurable? i.e. how do you get the most performance out of it?

Reply:
IMO most performance would be to not have many cells to check, in this case it would be 3^3 = 27